enhancements/microshift/multus-cni-for-microshift.md

```diff
@@ -0,0 +1,445 @@
+tracking-link:
+  - https://issues.redhat.com/browse/OCPSTRAT-473
+# see-also:
+#   - "/enhancements/this-other-neat-thing.md"
+# replaces:
+#   - "/enhancements/that-less-than-great-idea.md"
+# superseded-by:
+#   - "/enhancements/our-past-effort.md"
+---
+
+# Multus CNI for MicroShift
+
+## Summary
+
+Currently MicroShift ships [ovn-kubernetes](https://github.com/openshift/ovn-kubernetes) (ovn-k)
+CNI responsible for connectivity within and outside the cluster.
+There are users that have needs beyond what ovn-k offers like adding more interfaces to the Pods.
+Some example requirements are connecting Pods to the host's bridge interface or setting up complex networking based on VLAN.
+This functionality is Multus' trademark - adding additional interfaces to Pods.
+
+This enhancement explores providing Multus CNI as an optional component to MicroShift.
+
+## Motivation
+
+Providing Multus CNI for MicroShift will help users:
+- wanting to integrate MicroShift into existing environments
+- wanting to slowly, step by step, migrate to MicroShift without being required to refactor everything at once
+- wanting to add additional interfaces because of newly discovered requirements
+
+### User Stories
+
+* As a MicroShift admin, I want to add additional interfaces to certain Pods so that I can
+  make them accessible over networks that should not be available to rest of the cluster.
+* As a MicroShift admin, I want to add additional interfaces to certain Pods so that I can
+  access them directly (from outside the cluster) without using Kubernetes' networking such as
+  NodePorts, Load Balancers, Ingresses, etc.
+* As a MicroShift admin, I want to add additional interfaces to certain Pods so that I can
+  start slowly migrating an existing solution to MicroShift.
+
+### Goals
+
+- Provide optional Multus CNI for MicroShift clusters that can be added to existing clusters
+
+### Non-Goals
+
+- Automatically removing Multus from the cluster upon RPM uninstall
+- Support Multus for multi-node deployments of MicroShift
+- Providing network policies or admission webhook for Multus
+
+## Proposal
+
+Deliver Multus for MicroShift as an optional RPM containing required manifests that will be applied
+during MicroShift's start. There should be little to no changes to MicroShift itself as we want
+Multus CNI for MicroShift to fit optional components pattern.
+
+Manifests for deploying Multus on MicroShift will be based on existing manifests for OpenShift,
+but they will differ because OpenShift uses thick architecture Multus whereas MicroShift will use
+thin architecture Multus.
+
+How should cleanup of the Multus artifacts look like is an open question (see Open Questions section).
+
+### Workflow Description
+
+**User** is a human user responsible for setting up and managing Edge Devices.
+**Application** is user's workload that intends to use additional interfaces.
+
+1. User gathers all information about the networking environment of the edge device.
```

Comment:
The workflow doesn't include installing multus to an existing microshift device. Is it not going to be supported? If so, it might be good to be called out in the Goals section.
Installing multus to existing devices will change the node configuration (e.g. crio.conf.d) and requires reboot to take effect (pod can be recreated with multus, crio to take the new config).

Reply:
I think it should be supported with accompanying reboot.

> 1. User deploys the ostree commit onto the edge device.

This step covers adding Multus at any stage of lifecycle of the device using ostree system - regardless whether it's first or subsequent commit.

However, I'll make it more explicit

Reply:
Requiring a reboot after multus installation shall eliminate the issues I mentioned. +1